### PR TITLE
METRON-1073: Add option to limit fields returned from search query

### PR DESCRIPTION
**Pull request by merrimanr**
## Contributor Comments
This PR adds the option to explicitly declare which fields should be returned in search results.  Adding a "fields" property to the search request with a list of fields as the value should only return those fields (timestamp is always returned).  This is an example search request that only returns the "ip_src_addr" field:
```
{
  "fields":["ip_src_addr"],
  "from": 0,
  "indices": [
    "bro","snort"
  ],
  "query": "*",
  "size": 5
}
```

This can be tested in full dev through the Swagger UI.  Execute this search request in the search-controller and fields should be returned appropriately.

## Pull Request Checklist

Thank you for submitting a contribution to Apache Metron.  
Please refer to our [Development Guidelines](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=61332235) for the complete guide to follow for contributions.  
Please refer also to our [Build Verification Guidelines](https://cwiki.apache.org/confluence/display/METRON/Verifying+Builds?show-miniview) for complete smoke testing guides.  

In order to streamline the review of the contribution we ask you follow these guidelines and ask you to double check the following:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? If not one needs to be created at [Metron Jira](https://issues.apache.org/jira/browse/METRON/?selectedTab=com.atlassian.jira.jira-projects-plugin:summary-panel). 
- [x] Does your PR title start with METRON-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.
- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?

### For code changes:
- [x] Have you included steps to reproduce the behavior or problem that is being changed or addressed?
- [x] Have you included steps or a guide to how the change may be verified and tested manually?
- [x] Have you ensured that the full suite of tests and checks have been executed in the root metron folder via:
  ```
  mvn -q clean integration-test install && build_utils/verify_licenses.sh 
  ```

- [x] Have you written or updated unit tests and or integration tests to verify your changes?
- [x] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [x] Have you verified the basic functionality of the build by building and running locally with Vagrant full-dev environment or the equivalent?

### For documentation related changes:
- [x] Have you ensured that format looks appropriate for the output in which it is rendered by building and verifying the site-book? If not then run the following commands and the verify changes via `site-book/target/site/index.html`:

  ```
  cd site-book
  mvn site
  ```

#### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.
It is also recommended that [travis-ci](https://travis-ci.org) is set up for your personal repository such that your branches are built there before submitting a pull request.

### CODE REVIEW

File: metron-platform/metron-elasticsearch/src/main/java/org/apache/metron/elasticsearch/dao/ElasticsearchDao.java [hunk 1]
```diff
@@ -81,8 +82,14 @@ public SearchResponse search(SearchRequest searchRequest) throws InvalidSearchEx
             .size(searchRequest.getSize())
             .from(searchRequest.getFrom())
             .query(new QueryStringQueryBuilder(searchRequest.getQuery()))
-            .fetchSource(true)
+
             .trackScores(true);
+    Optional<List<String>> fields = searchRequest.getFields();
+    if (fields.isPresent()) {
+      searchSourceBuilder.fields(fields.get());
```
Comment (cestella) [inline metron-platform/metron-elasticsearch/src/main/java/org/apache/metron/elasticsearch/dao/ElasticsearchDao.java]: What happens if fields.get() is empty or null?

Comment (cestella) [inline metron-platform/metron-elasticsearch/src/main/java/org/apache/metron/elasticsearch/dao/ElasticsearchDao.java]: sorry, I should say, if fields is empty.

Comment (cestella) [inline metron-platform/metron-elasticsearch/src/main/java/org/apache/metron/elasticsearch/dao/ElasticsearchDao.java]: bah, I retract.  Sorry, I totally misread this.  Nevermind me ;)

File: metron-platform/metron-elasticsearch/src/main/java/org/apache/metron/elasticsearch/dao/ElasticsearchDao.java [hunk 2]
```diff
@@ -100,7 +107,16 @@ public SearchResponse search(SearchRequest searchRequest) throws InvalidSearchEx
     searchResponse.setResults(Arrays.stream(elasticsearchResponse.getHits().getHits()).map(searchHit -> {
       SearchResult searchResult = new SearchResult();
       searchResult.setId(searchHit.getId());
-      searchResult.setSource(searchHit.getSource());
+      Map<String, Object> source;
+      if (fields.isPresent()) {
+        source = new HashMap<>();
+        searchHit.getFields().forEach((key, value) -> {
+          source.put(key, value.getValues().size() == 1 ? value.getValue() : value.getValues());
```
Comment (mmiklavc) [inline metron-platform/metron-elasticsearch/src/main/java/org/apache/metron/elasticsearch/dao/ElasticsearchDao.java]: @merrimanr Do you have an integration test that checks when the result size is 0? I didn't notice one. I'm not familiar enough with `value.getValues().size()` to know how this works for boundary conditions. The ES javadoc doesn't really provide details.

Comment (merrimanr) [inline metron-platform/metron-elasticsearch/src/main/java/org/apache/metron/elasticsearch/dao/ElasticsearchDao.java]: If there are no results this line doesn't even get called because this code is inside a Stream map function that's iterating over the results.  The test case is really simple though so I added it just to be sure.

**Comment by mmiklavc** [comment]
+1 
Tested in full dev. Looks good @merrimanr, thanks for the contribution.
